network/src/protocols/discovery/mod.rs

```diff
@@ -84,8 +84,8 @@ pub const DISCOVERY_PROTOCOL_NAME: &[u8] = b"/libra/discovery/0.1.0";
 
 /// The actor running the discovery protocol.
 pub struct Discovery<TTicker, TSubstream> {
-    /// Note for self.
-    self_note: Note,
+    /// PeerId for self.
+    self_peer_id: PeerId,
```

Comment:
Hey, thanks for the contribution!

The reason why we store the "self" `Note` is in preparation for logic that changes the advertised `Note` while the validator is running. It looks like there is another useful change in here, though, which is storing our own `PeerId` as a field so we can avoid deserializing it again and again.

As such, it would helpful to keep the `self_note` field _and_ the `self_peer_id` field. I would also rename `self_peer_id` to just `peer_id` and (now that I think about it) `self_note` to just `note`. You might also have to prefix with an underscore `_note` since it will probably be unused and the linter will yell at you.

Reply:
Thanks for your review! Now I understand why we store the self `Note`. 💡  I'll update this PR.